app/actors/hyrax/actors/create_with_files_actor.rb

```diff
@@ -6,16 +6,26 @@ class CreateWithFilesActor < Hyrax::Actors::AbstractActor
       # @return [Boolean] true if create was successful
       def create(env)
         uploaded_file_ids = filter_file_ids(env.attributes.delete(:uploaded_files))
-        files = uploaded_files(uploaded_file_ids)
-        validate_files(files, env) && next_actor.create(env) && attach_files(files, env)
+        files             = uploaded_files(uploaded_file_ids)
+        # get a current copy of attributes, to protect against future mutations
+        attributes        = env.attributes.clone
```

Comment:
`#clone` creates a shallow copy of an `env.attributes` object, which means if the `attributes` object contains other objects, then those objects will be shared not copied so it still can be mutated.
 What is the structure of the `attributes`?   I'm not familiar with the codebase yet, maybe it is fine.

Reply:
in general, mutations to the attributes in the stack amount to addition/deletion of key/value pairs. still, it might not hurt to use `#deep_dup` just to be safe. i'll make the change.

Reply:
Yeah it's safer to use other approach.
I tired key/value pair (hash type). The original value can be mutated at deep level if using `#clone`:

```
irb(main):008:0> hsh1 = { k1: "a", k2: "b" }
=> {:k1=>"a", :k2=>"b"}
irb(main):009:0> hsh2 = hsh1.clone
=> {:k1=>"a", :k2=>"b"}
irb(main):010:0> hsh2[:k1].upcase!
=> "A"
irb(main):011:0> hsh1
=> {:k1=>"A", :k2=>"b"}
```

Reply:
But if it is only addition/deletion, it might be fine.